### PR TITLE
added draft new plasma

### PR DESCRIPTION
**Pull request by Shimwell**
This PR allows a plasma to be build from the 4 coordinates (top, botton, inner, outer). The 4 plasma shaping parameters are found from these coordinates and then used to build a regular plasma.

This plasma is needed to help make parametric reactors with a vertical build

### CODE REVIEW

File: paramak/parametric_components/tokamak_plasma_from_points.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import numpy as np
+import scipy
+
+from paramak import Plasma
+
+
+class Plasma(Plasma):
+    """Creates a double null tokamak plasma shape that is controlled
+       by 4 shaping parameters.
```
Comment (RemiTheWarrior) [inline paramak/parametric_components/tokamak_plasma_from_points.py:12]: Isn't it 3 parameters here ? Namely inner, outer and high point ? or does the hight point counted as 2 ?

File: paramak/parametric_components/tokamak_plasma_from_points.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import math
+from pathlib import Path
+
+import numpy as np
+import scipy
+
+from paramak import Plasma
+
+
+class Plasma(Plasma):
```
Comment (RemiTheWarrior) [inline paramak/parametric_components/tokamak_plasma_from_points.py:10]: Can a user choose between this Plasma (from points) and the inherited class Plasma (from parameters) ?

File: paramak/parametric_components/tokamak_plasma_from_points.py (hunk 1 above)
Comment (Shimwell) [reply paramak/parametric_components/tokamak_plasma_from_points.py:12]: I was thinking the high and low point are the same, but I guess this is not always true

File: paramak/parametric_components/tokamak_plasma_from_points.py (hunk 2 above)
Comment (Shimwell) [reply paramak/parametric_components/tokamak_plasma_from_points.py:10]: I need to finish, this PR but I like your plan of having one Plasma then two others plasma_form_points and plasma_from_parameters. I should rename this one

File: paramak/parametric_components/tokamak_plasma_from_points.py (hunk 1 above)
Comment (RemiTheWarrior) [reply paramak/parametric_components/tokamak_plasma_from_points.py:12]: They are symetrical indeed (in the case of an up-down symmetrical plasma). How do you wish to take vertical displacement into account ?

**Pull request merged**